languages/csharp/exercises/concept/basics/.docs/instructions.md

```diff
@@ -0,0 +1,43 @@
+In this exercise you're going to write some code to help you cook a brilliant lasagna from your favorite cooking book.
+
+You have four tasks, all related to the time spent cooking the lasasgna.
+
+### 1. Define the expected oven time in minutes
+
+Define the `Lasagna.ExpectedMinutesInOven` method that does not take any parameters and returns how many minutes the lasagna should be in the oven. According to the cooking book, the expected oven time in minutes is 40:
```

Comment:
`Lasagna#ExpectedMinutesInOven` etc

Reply:
Do you mean to indicate that we are referring to an instance method? If so, I don't think I've ever seen the `#` character being used in C# (I think that is your Ruby background speaking :)). For C#, I believe the `.` is also used for instance methods.

To verify this. I've created a small class:

```csharp
class Test
{
    void Hello()
    {
        throw new NotImplementedException();
    }
```

If the `Hello` method is called, the following is shown in the stack trace:

```
Unhandled exception. System.NotImplementedException: The method or operation is
not implemented.
   at Test.Hello()
```
    }

Reply:
In text, I've always seen class methods as `Class.method` and instance methods as `Class#method`. It's not actually my Ruby background (for once! :)) - I was taught it at Uni and seen in it the wild lots since. If you use the `.` for both, how can you tell whether someone is discussing an instance method or a class method in a spec?

Reply:
Hmmm, maybe it _is_ me, but I've not seen it used in the C# context. I'm not necessarily opposed to it, but I would like to use what is customary for C#. Let's ask @mikedamay @robkeim and @wolf99 for their opinion.

Reply:
My 2c: 
I've not seen the hash convention used before in C#, and I don't recall seeing it in documentation. Further, this SO answer seems to provide ample answers that the octothorpe use in Ruby was made specifically for Ruby https://stackoverflow.com/questions/736120/why-are-methods-in-ruby-documentation-preceded-by-a-hash-sign. There is precedence in other languages it appears (for other purposes, JavaDocs in Java, some mention of it in SmallTalk, etc), but these occur for different purposes and independent to those languages rather than as a cross language convention. So although I can't say for definite, the evidence of my brief research does side with it not applying to C#. 

To specifically answer your question @iHiD, in languages like this I was taught that the spec for the method is the method definition - for class methods this includes the modifiers and parameters, so `static` is shown, e.g. `public static myStaticFoo(int bar);`. When reading in prose, the reader is expected to have the definition to hand, or for the modifiers to be otherwise described or explained in the text. I can't speak for Ruby, from my own experience this is fairly standard for languages that are more C-ish.

Reply:
@iHiD As @wolf99 has had the same experience as I had, I'm gonna keep it the way it is :)